msfconsole

```diff
@@ -8,6 +8,9 @@
 require 'pathname'
 
 begin
+  # TODO: Temporary until warnings can be turned on only for developers to prevent confusion for end users
+  Warning[:deprecated] = false
```

Comment:
Would any of the other tools be impacted, like msfvenom etc?

Reply:
I don't believe this should affect any other tools no
do you think we should be disabling warnings from other tools? I don't think we've had any issues from that (yet)

Reply:
In some scenarios msfvenom loads msf, which will trigger warnings:

```
$ ruby --version
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-darwin18]
$ bundle exec ruby ./msfvenom -p windows/meterpreter/reverse_tcp LHOST=127.0.0.1 --encrypt rc4 --encrypt-key thisisakey -f c
/Users/user/.rvm/gems/ruby-2.7.0/gems/activerecord-4.2.11.1/lib/active_record/connection_adapters/abstract_adapter.rb:84: warning: deprecated Object#=~ is called on Integer; it always returns nil
/Users/user/.rvm/gems/ruby-2.7.0/gems/activerecord-4.2.11.1/lib/active_record/connection_adapters/abstract_adapter.rb:84: warning: deprecated Object#=~ is called on Integer; it always returns nil

...
```

Other tools may also load msf

Reply:
I'm good to land this change for now, and we can look into the other edgecases as part of the follow up work :+1:

Reply:
```
➜  metasploit-framework git:(turn-off-warnings) ✗ ./msfvenom -p windows/meterpreter/reverse_tcp LHOST=127.0.0.1 --encrypt rc4 --encrypt-key thisisakey -f c
/Users/dwelch/.rvm/gems/ruby-2.7.0@metasploit-framework/gems/activerecord-4.2.11.1/lib/active_record/connection_adapters/abstract_adapter.rb:84: warning: deprecated Object#=~ is called on Integer; it always returns nil
/Users/dwelch/.rvm/gems/ruby-2.7.0@metasploit-framework/gems/activerecord-4.2.11.1/lib/active_record/connection_adapters/abstract_adapter.rb:84: warning: deprecated Object#=~ is called on Integer; it always returns nil
```
just checked, it doesn't affect msfvenom, it's loading those files sure but not via the msfconsole file where I added the change

Reply:
msfvenom loads msf directly here:

https://github.com/rapid7/metasploit-framework/blob/2dc26db9e1c2beb044d2cefd5b0839382561f265/msfvenom#L21-L32

I think that means that either the deprecated warning suppression needs be in the entrypoints of our tools, or closer to the entrypoint of msf itself.